### ISSUE TITLE
While deploying Zalando getting issue-  could not create cluster: could not create pod disruption budget

### ISSUE THREAD

**Issue by Deepanshu6dec** [zalando/postgres-operator#2098]
- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.8.2-26-gacb3ffd7
- **Where do you run it - cloud or metal? Kubernetes**  Bare Metal K8s]
- **Are you running Postgres Operator in production?** [no]
- **Type of issue?** [Bug report, question, feature request, etc.]

Hi Team 

We are deploying zalando with the help of helm and it was working fine in our environment earlier this month. Zalando is having some issues when deploying. It is showing an error:

 ```could not create cluster: could not create pod disruption budget: the server could not find the requested resource" cluster-name=default/c-cluster pkg=controller worker=0```

The pod distribution budget policy is deprecated in the V1.25 kubernetes version found in zalando.

We have tried installing the k8s cluster with 1.24.0 but are still getting the same issue.

Please find the error:

```
# Postgres Operator logs:

  Normal   Create     18s   postgres-operator  Started creation of new cluster resources
  Normal   Endpoints  18s   postgres-operator  Endpoint "default/c-cluster" has been successfully created
  Normal   Services   18s   postgres-operator  The service "default/c-cluster" for role master has been successfully created
  Normal   Services   18s   postgres-operator  The service "default/c-cluster-repl" for role replica has been successfully created
  Normal   Secrets    17s   postgres-operator  The secrets have been successfully created
  Warning  Create     17s   postgres-operator  could not create cluster: could not create pod disruption budget: the server could not find the requested resource

```

Kubernetes version

```
k8smn    Ready      control-plane,master   39d   v1.24.0
k8swn1   Ready      worker                         39d   v1.24.0
k8swn2   Ready      worker                         39d   v1.24.0

```
Operator Logs:

```
time="2022-11-01T07:01:55Z" level=info msg="creating a new Postgres cluster" cluster-name=default/c-cluster pkg=controller worker=0
time="2022-11-01T07:01:55Z" level=warning msg="master is not running, generated master endpoint does not contain any addresses" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=info msg="endpoint \"default/c-cluster\" has been successfully created" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=info msg="master service \"default/c-cluster\" has been successfully created" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=info msg="replica service \"default/c-cluster-repl\" has been successfully created" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=debug msg="team API is disabled" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=debug msg="team API is disabled" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=info msg="users have been initialized" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:55Z" level=info msg="syncing secrets" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:56Z" level=info msg="secrets have been successfully created" cluster-name=default/c-cluster pkg=cluster worker=0
time="2022-11-01T07:01:56Z" level=error msg="could not create cluster: could not create pod disruption budget: the server could not find the requested resource" cluster-name=default/c-cluster pkg=controller worker=0
```

With the same configuration, it was working fine. Can you please help to fix this issue

**Comment by dmvolod** [comment]
Latest operator version v1.8.2 is not supported on Kubernetes v1.24 and later.
This issue already been fixed in #2008 and will be available in next release.

**Comment by Deepanshu6dec** [comment]
May I know when we expect the next release to fix this issue?

**Comment by dmvolod** [comment]
@Deepanshu6dec there is no due date for the 1.9 milestone yet https://github.com/zalando/postgres-operator/milestone/12
Please ask this question to the main contributors https://github.com/zalando/postgres-operator/blob/master/CODEOWNERS

**Comment by Deepanshu6dec** [comment]
@FxKu Can you please help us here

**Comment by alairock** [comment]
@dmvolod is there a workaround for the time being?

**Comment by dmvolod** [comment]
@alairock the only workaround I can see now is to build the Docker operator image from the `master` branch, since that issue has already been fixed.

**Comment by Deepanshu6dec** [comment]
@dmvolod I have seen the docker file is in docker  in https://github.com/zalando/postgres-operator/tree/master/docker
Shall be directly build it with docker command and test it out locally in our environment

**Comment by Deepanshu6dec** [comment]
Issue is remain same, Lets wait for the release from Zalando team

**Comment by panpan0000** [comment]
Long time before last 1.8.2.
Looking forward to the 1.9.0 very soon.
kubernetes 1.25 has been wildly used.

 @sdudoladov @Jan-M @FxKu @jopadi @idanovinda @hughcapet

**Comment by benedikt-bartscher** [comment]
Same here, would be awesome if you release a new version soon!

**Comment by cro** [comment]
Agreed, I do not have access to any pre-1.24 k8s clusters at all.

**Comment by tommyJimmy87** [comment]
For me is working after building the docker image from the master branch directly, is not the optimal solution at the moment, but while waiting for the new release I have it working

**Comment by elliottneilclark** [comment]
Regular releases and an easy release process are pretty important for any infrastructure project. Is there something that we as the community can do to help with this work? (releasing regularly, building on other arches, etc)

**Comment by aman-quartic** [comment]
@tommyJimmy87 Thanks. If someone is looking for a more straightforward temporary solution, use an older version of k3s. `v1.23.14+k3s1` worked for me.
`curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v1.23.14+k3s1 sh -`

**Comment by katlol** [comment]
A quick look at https://registry.opensource.zalan.do/v1/repositories/acid/postgres-operator/tags shows that Zalando is building images for each commit.

This Kustomize patch works on 1.26.0:
```
images:
  - name: registry.opensource.zalan.do/acid/postgres-operator
    newTag: v1.8.2-43-g3e148ea5
```

**Comment by kskalski** [comment]
For chart users, this change to the newer image solves the problem:

```
+++ b/charts/postgres-operator/values.yaml
@@ -1,7 +1,7 @@
 image:
   registry: registry.opensource.zalan.do
   repository: acid/postgres-operator
-  tag: v1.8.2
+  tag: v1.8.2-43-g3e148ea5
```

**Comment by MohamedShaj** [comment]
> Latest operator version v1.8.2 is not supported on Kubernetes v1.24 and later. This issue already been fixed in #2008 and will be available in next release.

when is the next release available and any drawback in v1.8.2 for the production-ready cluster?

for v1.25.5+k3s1 what is the perfect and stable version in postgres-operator ?

**Comment by dmvolod** [comment]
Hi @MohamedShaj 
1.9 is coming next few days https://github.com/zalando/postgres-operator/milestone/12

**Comment by ChuckPerry** [comment]
Hi, Team
we are facing the same issue 
`could not create cluster: could not create pod disruption budget: the server could not find the requested resource" cluster-name=postgres/business-application-hosting pkg=controller worker=1`

Our kube version is v1.19.2 and operator is 1.9.0.

**Comment by dmvolod** [comment]
Hi @ChuckPerry 
You've got incompatibility case discussed in #2094
This means, that for operator v1.9.x supported Kubernetes version, will be 1.23.x ± 1 release, but seems to will work starting v1.21.x (policy/v1 API version, available since this version)

**Comment by ChuckPerry** [comment]
> 

Got it. Thanks a lot.

**Comment by ChuckPerry** [comment]
> Hi @ChuckPerry You've got incompatibility case discussed in #2094 This means, that for operator v1.9.x supported Kubernetes version, will be 1.23.x ± 1 release, but seems to will work starting v1.21.x (policy/v1 API version, available since this version)

Thanks a lot.

**Comment by yajo** [comment]
[1.9.0 is released](https://github.com/zalando/postgres-operator/releases/tag/v1.9.0). Should this be closed then?

**Comment by provokateurin** [comment]
At least for me the problem is solved with 1.9.0

**Comment by amirkusedghi** [comment]
Hi all, 
Following on this, I have Postgres-operator v1.5 on Kubernetes v1.23, I have an upcoming upgrade of Kubernetes to v1.24? Should I expect disruption? if so what version I should upgrade to as my currently deployed is pretty old?

**Comment by FxKu** [comment]
@amirkusedghi read the breaking changes section in the notes of each release. Ideally you should update to the newest version or 1.9.0. Thinking o Postgres versions that were dropped and changes in LB DNS names. You can also expect rolling updates to happen with your Postgres clusters.

I will close this ticket now because it was solved with 1.9.0
